src/coreclr/jit/codegenarm.cpp

```diff
@@ -203,15 +203,15 @@ void CodeGen::instGen_Set_Reg_To_Imm(emitAttr  size,
 
             if (GetEmitter()->isLowRegister(reg) && (imm_hi16 == 0xffff) && ((imm_lo16 & 0x8000) == 0x8000))
             {
-                GetEmitter()->emitIns_R_R(INS_sxth, EA_4BYTE, reg, reg);
+                GetEmitter()->emitIns_Mov(INS_sxth, EA_4BYTE, reg, reg, /* canSkip */ false);
```

Comment:
Most of these `emitIns_Mov` calls should "probably" be `inst_Mov` or `inst_Move_Extend` calls instead, where the instruction is picked based on the register and type.